frappe/core/doctype/user/user.py

```diff
@@ -811,6 +812,20 @@ def check_roles_added(self):
 			},
 		)
 
+	def validate_ip_addr(self):
+		# remove whitespace
+		if not self.restrict_ip:
+			return
+		if str.isspace(self.restrict_ip):
+			self.restrict_ip = None
+		else:
+			if self.restrict_ip.find(",") != -1:
+				ip_string = ""
+				for s in self.restrict_ip.split(","):
+					s.strip()
+					ip_string = s + ","
+				self.restrict_ip = ip_string
```

Comment:
There is some code duplication here. You just need to:

1. Strip whitespace at ends using `str.strip()`, checking for whitespace isn't necessary. `""` and `None` both are falsy and it's fine for us to treat them interchangeably. 
2. Parse them with existing function `self.get_restricted_ip_list` which will also strip whitespace in middle. 
3. Rejoin your parse list `",".join(parsed_list)`


Offtopic: Familiarize yourself with stdlib to avoid re-implementing basic things like `str.join`